specs/plugin-extensibility.md

```diff
@@ -20,15 +20,41 @@ Keys and associated certificates used for signing artifacts using Notation could
   * Notation MUST work with a plugin that implements a matching or lower minor version of the plugin contract. Notation SHALL NOT support using a plugin with higher version of plugin contract.
   * A plugin MUST support a single plugin contract version, per major version.
 
-Notation will invoke plugins as executable, pass parameters using command line arguments, and use standard IO streams to pass request/response payloads. This mechanism is used as Go language (used to develop [Notation Go library](https://github.com/notaryproject/notation-go)) does not have a [in built support](https://github.com/golang/go/issues/19282) to load and execute plugins that works across OS platforms. Other mechanisms like gRPC require every plugin to be implemented as a service/daemon.
+Notation will invoke plugins as executable, pass parameters using command line arguments, and use standard IO streams to pass request/response payloads.
+
+### Package a plugin
+
+Plugin publisher SHOULD follow these conventions to package and distribute plugin files:
+
+* The file formats supported by Notation include `.zip`, `.tar.gz`, and single plugin executable file. It is highly recommended to package the single plugin executable file into an archive for installation security and efficiency consideration
+* Plugin executable file MUST follow the naming convention `notation-{plugin-name}` 
+* Notation supports installing a plugin from an `https` URL or from a file in the filesystem
+* If the archive format is `.zip` or `.tar.gz`, there MUST be one and only one plugin executable file within each archive
+* Plugin publisher SHOULD provide SHA256 checksum for each released archive 
+* If the plugin has external dependencies, plugin publisher MUST include them in the archive
+* License file is optional but it's recommended to be included in the archive
+* To avoid [ZIP bombs](https://en.wikipedia.org/wiki/Zip_bomb), plugin archive file size is limited to less than 256 MiB. The sum of the extracted files is also limited to 256 MiB
```

Comment:
If the release format is `single plugin executable file`, it also has the 256 MiB size limit.